### ISSUE TITLE
Demo app fails to build --prod with UglifyJs error

### ISSUE THREAD

**Issue by justingrayston** [tinesoft/generator-ngx-library#65]
### Bug Report or Feature Request (mark with an `x`)
```
- [ x ] bug report -> please search issues before submitting
- [ ] feature request
```

### Node / Npm or Yarn Version
macOS Sierra

Output from: node --version= v6.11.1,
Output from: yo --version = 2.0.0
Output from: yarn --version = 0.27.5
### Targeted Angular version
4.x.x

### Repro steps.
Generate a new library go to the demo and run `ng build --prod`

### The log given by the failure.
```
ERROR in main.dd7e42304148e7406ffc.bundle.js from UglifyJs
Unexpected token: name (LibComponent) [main.dd7e42304148e7406ffc.bundle.js:18445,6]
```
### Desired functionality.
It should build.

### Mention any other details that might be useful.
This error appears mentioned in quite a few places on Github. Note https://github.com/webpack/webpack/issues/2972 which would suggest that at compile angular-cli / webpack is using `my-ngx-library.js` and not the umd bundle.

This really is a great fully featured generator and really needed, but this is blocking me. Every library module I have created with the generator suffers from this. I am not sure what the problem is atm.

**Comment by justingrayston** [comment]
I think it maybe related to the module being linked https://github.com/babel/babel-loader/issues/149

**Comment by justingrayston** [comment]
Actually it doesn't seem to be that. 

I have pushed the templated generated library to here https://github.com/justingrayston/generated-ngx-library-test and also created a distribution repo here https://github.com/justingrayston/generated-ngx-library-test-dist

If I make sure the package isn't linked, and `yarn add git+https://github.com/justingrayston/generated-ngx-library-test-dist.git` the package is added and you can see in node_modules. It still fails to build with the `--prod` flag.

**Comment by tinesoft** [comment]
Hi @justingrayston,

Thanks for reporting this issue 👍 , and for your investigations.
There was a bug in distributed `package.json` of built library when targetting Angular >=4: the `es2015`version of the JS bundle was referred to under `"module"` instead of the `es5` one.

I've just released [v3.1.0](https://github.com/tinesoft/generator-ngx-library/releases/tag/v3.1.0) with that fix (among others)

To get rid of the error in your existing libraries,  simply:
- re-rerun the generator once updated or manually edit the `gulpfile.js`as show in fd0d366dc8c370c7ac12fa8ea2b9f139e59ffc49 
-re-build or re-publish your library, it should correctly set the module to use in `package.json`

Enjoy!

**Comment by justingrayston** [comment]
Thanks Tine!
